xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
It's really a good idea using a single thread for all domids, 
I think we don't need to put
` ``let X ()= try qmp_read ~in_event_loop:true```
 inside the `next` function.
We can call `qmp_read` after sending qmp command. Like:

```
let rec qmp_read domid c =
	let qmp_msg = Qmp_protocol.read c in
	match qmp_msg with
	| Error _ | Success (_, _) -> qmp_msg
	| Event event -> QMP.Events.add domid event; qmp_read domid c
	| _ -> raise (QMP_receive_error (Printf.sprintf "Get unexpected Qmp message: %s" (Qmp.string_of_message qmp_msg)))
```